cmd/crdvalidator/README.md

```diff
@@ -0,0 +1,96 @@
+# crdvalidator
+
+## Summary
+A part of the core value proposition for RukPak is the safe upgrade of a [CustomResourceDefinition](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/) (`CRD`). As a result, this repository comes equipped with a "CRD Validator Webhook" (`crdvalidator`), that will validate all `CRD` upgrades created by RukPak. This protects your `BundleInstance` pivots from having potentially dangerous effects such as data loss. 
+
+### How to use
+
+`crdvalidator` will ensure that any `CRD` upgrade is safe as long as it is owned by RukPak. If you would like to disable validation for a `CRD` included in a specific `Bundle`, you will need to explicitly set the `core.rukpak.io/safe-upgrade-validation` annotation to `disabled` within that specific `Bundle`.
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    core.rukpak.io/safe-upgrade-validation: disabled # <--
+```
+
+### What is protected
+There are three main cases `crdvalidator` is checking against with every specified `CRD` upgrade. These are where a new `CRD`:
+1. Removes a stored version and ensures that removing the stored version does not result in data loss.
+2. Changes a version that old `CRD` had and it must validate existing CRs against the new schema.
+3. Adds a version that old `CRD` does not have. In this case `crdvalidator` checks if the conversion strategy is:
+    - None, then ensures that existing CRs validate with new schema.
+    - Webhook, then allow update (assuming webhook handles conversion correctly)
+
+#### Update the schema that invalidates existing CRs
+Say that you have installed a `CRD` onto the cluster with a Kind of `Sample` that has a single property of `examplearray`. 
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.7.0
+    core.rukpak.io/safe-upgrade-validation: enabled
+  creationTimestamp: null
+  name: templates.combo.io
+#
+# ...
+#
+spec:
+    versions:
+    - name: v1alpha1
+        schema:
+        openAPIV3Schema:
+            properties:
+                examplearray:
+                    description: Examplearray is an array of strings to demonstrate crd validation logic.
+                    type: array
+                    minItems: 1
+                    items:
+                    type: string
+```
+
+With this installed onto the cluster, you go ahead and create a `CR` of a `Sample` onto the cluster. If you were to then update that `CRD`'s schema in such a way that that `CR` you created was invalidated, like so:
```

Comment:
should we actually show a `Sample` manifest here, or is this just for the sake of example? Is there any intent that the user would actually follow this example?

Reply:
That would probably make this a bit clearer and flow more naturally. I'll add that.